### PR TITLE
Order confirmation and preparation

### PR DESCRIPTION
**Pull request by caressahh**
### The following features were added to the app ##
- Confirmation view
- Add input for number of boxes if customer has selected delivery
- Buttons for Gigafood to confirm or cancel the order.
- Set order status to confirmed/canceled depending on selection.
- Sending an email to the customer with order confirmation or cancellation.

We used ActiveAdmin to add Order feature. The implementation simplifies the order administration for GigaFood and cuts manual tasks. 

### What we learned ###
- How to use ActiveAdmin
- How to use ActionMailer
- Use Cucumber to test emails
- Use RSpec to unit test emails and ActionMailer functionality

### CODE REVIEW

File: features/orders.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
+Feature: List orders
+  As Giga-Food
+  In order to make a decision to start an order
+  I like to have three options approve, cancel and a input filed with transport size
+
+  Background:
+    Given an admin exists with email "admin@example.com" and password "password"
+    And I'm loged in as admin user "admin@example.com"
+    Given I go to the dashboard
```
Comment (satorinic) [inline features/orders.feature]: Change to And?
In the test there are several inconsistent step. Check again.

"Given" for background steps
"When" for actions 
"Then" for assertions
Use "And" for repetting the same type of step

File: spec/factories/orders.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  factory :order do
+    name "MyString"
```
Comment (satorinic) [inline spec/factories/orders.rb]: change to real values?

File: spec/factories/orders.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    description "MyText"
+    price 1
+    order_date "2017-09-09 10:15:58"
+    delivery_date "2017-09-09 10:15:58"
+    address "MyText"
```
Comment (satorinic) [inline spec/factories/orders.rb]: Change to real values?

File: spec/models/order_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (satorinic) [inline spec/models/order_spec.rb]: Should this be deleted? Or will you write something to test?

**Review by satorinic — COMMENTED** [review]
Looks good. Having said that I propose some modifications in the test file and in the factories

File: features/orders.feature (hunk 1 above)
Comment (faraznaeem) [reply features/orders.feature]: Done

File: spec/factories/orders.rb (hunk 3 above)
Comment (faraznaeem) [reply spec/factories/orders.rb]: Done

File: spec/factories/orders.rb (hunk 2 above)
Comment (faraznaeem) [reply spec/factories/orders.rb]: done

**Comment by faraznaeem** [comment]
Gracias

**Pull request merged**